### PR TITLE
Entity and Goal Mappings

### PR DESCRIPTION
**Pull request by MaheEvil**
Opinions needed for name of newly mapped class `MobEntityActiveTargetGoal` (`C_16xvdchlc`)
- The class is essentially a variation of ActiveTargetGoal but for Non Path Finding mob entities (only used for slime in 1.8 afaik)

### CODE REVIEW

**Comment by SpaceWalkerRS** [comment]
Then there is also `C_94dijdhgf` which seems to be a variation that takes scoreboard teams into account. Both of these variations seem to me to be more basic versions, with the only real difference being that there's fewer calculations in the target filters?

**Comment by InspectorBoat** [comment]
`SimpleActiveTargetGoal`?

**Comment by MaheEvil** [comment]
> Then there is also `C_94dijdhgf` which seems to be a variation that takes scoreboard teams into account. Both of these variations seem to me to be more basic versions, with the only real difference being that there's fewer calculations in the target filters?

I currently have that named as `MobEntityPlayerTargetGoal` temporarily, as it's filter is hardcoded to only accept PlayerEntities  
image

**Comment by SpaceWalkerRS** [comment]
Perhaps the following could work: `C_64qcggszu` -> `NearestAttackableTargetGoal`, `C_16xvdchlc` -> `NearestTargetGoal`, `C_94dijdhgf` -> `NearestAttackablePlayerGoal`.

Also a few other comments:
- Generally, use `pos` not `blockPos`, `state` not `blockState`, `stack` not `itemStack`, `client` not `mcClient`.
- Generally, don't put the type of a field/method in its name: `getFishLoot` not `getFishLootList`, `target` not `targetEntity`, `mob` not `mobEntity`, etc. There are exceptions but most of the time these type descriptors are redundant and could (or should) be replaced by a descriptor of what the field/method is used for (e.g. `followTarget` or `attackTarget` rather than `targetEntity`).

**Comment by MaheEvil** [comment]
> Perhaps the following could work: `C_64qcggszu` -> `NearestAttackableTargetGoal`, `C_16xvdchlc` -> `NearestTargetGoal`, `C_94dijdhgf` -> `NearestAttackablePlayerGoal`.
> 
> Also a few other comments:
> 
>     * Generally, use `pos` not `blockPos`, `state` not `blockState`, `stack` not `itemStack`, `client` not `mcClient`.
> 
>     * Generally, don't put the type of a field/method in its name: `getFishLoot` not `getFishLootList`, `target` not `targetEntity`, `mob` not `mobEntity`, etc. There are exceptions but most of the time these type descriptors are redundant and could (or should) be replaced by a descriptor of what the field/method is used for (e.g. `followTarget` or `attackTarget` rather than `targetEntity`).

Alright, I'll go and edit them, and thanks for the tips

**Comment by MaheEvil** [comment]
And about the names for the goals, Yea those names sounds alright.

Although I am not exactly sure about the two goals being `NearestTargetGoal` and `NearestAttackableTargetGoal`, it would sure be an improvement over the current names of ActiveTargetGoal, but still both in a sense try to find "attack-able" target mobs

**Comment by SpaceWalkerRS** [comment]
> Although I am not exactly sure about the two goals being `NearestTargetGoal` and `NearestAttackableTargetGoal`, it would sure be an improvement over the current names of ActiveTargetGoal, but still both in a sense try to find "attack-able" target mobs

That's a fair point. I struggle to come up with good names that don't just boil down to adding `ForPathAwareMobs` to one of them, which sounds very clunky.

**Comment by MaheEvil** [comment]
> > Although I am not exactly sure about the two goals being `NearestTargetGoal` and `NearestAttackableTargetGoal`, it would sure be an improvement over the current names of ActiveTargetGoal, but still both in a sense try to find "attack-able" target mobs
> 
> That's a fair point. I struggle to come up with good names that don't just boil down to adding `ForPathAwareMobs` to one of them, which sounds very clunky.

Yes, We can either use a really clunky and long name, or have a decent enough name with java-docs to clarify.
Since I am struggling to even come with a name that is clunky but clear, will probably have to go with the latter option unless someone else has a better suggestion

**Comment by SpaceWalkerRS** [comment]
My vote goes to changing them up. I'm not too fond of the current name.

File: mappings/1.3.tiny [hunk 1]
```diff
@@ -10406,8 +10406,8 @@ c	net/minecraft/C_32myydzeb	net/minecraft/entity/ItemEntity
 		p	1		world
 		p	2		x
 		p	4		y
-		p	6		p_6
-		p	8		p_8
+		p	6		z
+		p	8		itemStack
```
Comment (SpaceWalkerRS) [inline mappings/1.3.tiny:10410]: `stack`

File: mappings/1.4.5#12w49a.tinydiff [hunk 2]
```diff
@@ -362,8 +362,8 @@ c	net/minecraft/C_52beyppli		net/minecraft/entity/FireworksEntity
 		p	1			world
 		p	2			x
 		p	4			y
-		p	6			p_6
-		p	8			p_8
+		p	6			z
+		p	8			itemStack
```
Comment (SpaceWalkerRS) [inline mappings/1.4.5#12w49a.tinydiff:366]: `stack`

File: mappings/1.7.4#14w02a.tinydiff [hunk 3]
```diff
@@ -6109,9 +6109,9 @@ c	net/minecraft/C_47ldwddrb
 		p	2			y
 	m	(DZLnet/minecraft/C_68zcrzyxg;Lnet/minecraft/C_76varpwca;)V	m_64xomyper		onFall
 		p	1			dy
-		p	3			p_3
-		p	4			p_4
-		p	5			p_5
+		p	3			landed
+		p	4			block
+		p	5			blockPos
```
Comment (SpaceWalkerRS) [inline mappings/1.7.4#14w02a.tinydiff:6114]: `pos`

File: mappings/14w05b#14w06a.tinydiff [hunk 4]
```diff
@@ -971,46 +971,46 @@ c	net/minecraft/C_66oqmtrvn
 	m	(Lnet/minecraft/C_03obpszvn;Lnet/minecraft/C_14rixalrn;)Lnet/minecraft/C_14rixalrn;	m_10qsdmozm		m_10qsdmozm
 		p	1			p_1
 		p	2			p_2
-c	net/minecraft/C_66oqmtrvn$C_10gmowcxq		C_10gmowcxq
-	f	Lnet/minecraft/C_66oqmtrvn;	f_82tfscilk		f_82tfscilk
+c	net/minecraft/C_66oqmtrvn$C_10gmowcxq		MoveTowardsTargetGoal
+	f	Lnet/minecraft/C_66oqmtrvn;	f_82tfscilk		slimeEntity
```
Comment (SpaceWalkerRS) [inline mappings/14w05b#14w06a.tinydiff:975]: `sime`

File: mappings/14w05b#14w06a.tinydiff [hunk 5]
```diff
@@ -971,46 +971,46 @@ c	net/minecraft/C_66oqmtrvn
...
+c	net/minecraft/C_66oqmtrvn$C_21suglfgj		SwimAroundGoal
+	f	Lnet/minecraft/C_66oqmtrvn;	f_02frhgwid		entity
 	m	(Lnet/minecraft/C_66oqmtrvn;)V	<init>		<init>
-		p	1			p_1
+		p	1			entity
 	m	()Z	m_90hiwtytq		m_90hiwtytq
 	m	()V	m_89nkhezid		m_89nkhezid
-c	net/minecraft/C_66oqmtrvn$C_24nqbpjus		C_24nqbpjus
-	f	Lnet/minecraft/C_66oqmtrvn;	f_37ugsdzcs		f_37ugsdzcs
+c	net/minecraft/C_66oqmtrvn$C_24nqbpjus		LeapTowardsTargetEntityGoal
```
Comment (SpaceWalkerRS) [inline mappings/14w05b#14w06a.tinydiff:989]: `LeapTowardsTargetGoal` is more consistent with the other target goals

File: mappings/14w05b#14w06a.tinydiff [hunk 6]
```diff
@@ -971,46 +971,46 @@ c	net/minecraft/C_66oqmtrvn
...
 	m	()V	m_89nkhezid		m_89nkhezid
-c	net/minecraft/C_66oqmtrvn$C_39nulntrs		C_39nulntrs
+c	net/minecraft/C_66oqmtrvn$C_39nulntrs		MovementControl
 	f	F	f_90hjxvrag		f_90hjxvrag
-	f	I	f_94hjovssz		f_94hjovssz
-	f	Lnet/minecraft/C_66oqmtrvn;	f_22qaxyavr		f_22qaxyavr
-	f	Z	f_09gnkzzhn		f_09gnkzzhn
+	f	I	f_94hjovssz		jumpCooldownTicks
+	f	Lnet/minecraft/C_66oqmtrvn;	f_22qaxyavr		entity
+	f	Z	f_09gnkzzhn		hasTargetEntity
```
Comment (SpaceWalkerRS) [inline mappings/14w05b#14w06a.tinydiff:1001]: `hasTarget`

File: mappings/14w05b#14w06a.tinydiff [hunk 7]
```diff
@@ -1479,19 +1479,19 @@ c	net/minecraft/C_78ehokafe		net/minecraft/C_78ehokafe
...
-	f	Lorg/apache/logging/log4j/Logger;	f_38kqktowf		f_38kqktowf
-	f	Lnet/minecraft/C_81psrrogw;	f_23xjluxdo		f_23xjluxdo
+c	net/minecraft/C_94dijdhgf		net/minecraft/entity/ai/goal/MobEntityPlayerTargetGoal
+	f	Lorg/apache/logging/log4j/Logger;	f_38kqktowf		LOGGER
+	f	Lnet/minecraft/C_81psrrogw;	f_23xjluxdo		mob
 	f	Lnet/minecraft/C_52ygecrnn;	f_23hectams		f_23hectams
-	f	Lnet/minecraft/C_64qcggszu$C_11itnlavb;	f_53mpccbyu		f_53mpccbyu
-	f	Lnet/minecraft/C_97zulxhng;	f_54kjtdzlp		f_54kjtdzlp
+	f	Lnet/minecraft/C_64qcggszu$C_11itnlavb;	f_53mpccbyu		entityDistanceComparator
+	f	Lnet/minecraft/C_97zulxhng;	f_54kjtdzlp		targetEntity
```
Comment (SpaceWalkerRS) [inline mappings/14w05b#14w06a.tinydiff:1487]: `target`

File: mappings/14w10c#14w11a.tinydiff [hunk 8]
```diff
@@ -1100,14 +1100,14 @@ c	net/minecraft/C_89dwttwmw
 		p	5			p_5
 c	net/minecraft/C_94dijdhgf
 	f	Lnet/minecraft/C_52ygecrnn;	f_23hectams	f_23hectams	
-	f	Lcom/google/common/base/Predicate;	f_23hectams		f_23hectams
+	f	Lcom/google/common/base/Predicate;	f_23hectams		targetEntityFilter
```
Comment (SpaceWalkerRS) [inline mappings/14w10c#14w11a.tinydiff:1103]: `targetFilter`

File: mappings/14w27b#14w28a.tinydiff [hunk 9]
```diff
@@ -1406,7 +1406,7 @@ c	net/minecraft/C_51ouvctku
 c	net/minecraft/C_51wlzxfql
 	f	Lnet/minecraft/C_68zcrzyxg;	f_82afhplds	block	
 	f	I	f_37nwdqnog	metadata	
-	f	Lnet/minecraft/C_17agfiprw;	f_63iwqbtzw		f_63iwqbtzw
+	f	Lnet/minecraft/C_17agfiprw;	f_63iwqbtzw		blockState
```
Comment (SpaceWalkerRS) [inline mappings/14w27b#14w28a.tinydiff:1409]: `state`

File: mappings/14w27b#14w28a.tinydiff [hunk 10]
```diff
@@ -1425,8 +1425,8 @@ c	net/minecraft/C_51wlzxfql
 		p	1			world
 		p	2			x
 		p	4			y
-		p	6			p_6
-		p	8			p_8
+		p	6			z
+		p	8			blockState
```
Comment (SpaceWalkerRS) [inline mappings/14w27b#14w28a.tinydiff:1429]: `state`

File: mappings/14w27b#14w28a.tinydiff [hunk 11]
```diff
@@ -2623,21 +2623,22 @@ c	net/minecraft/C_16bwpyyiv		net/minecraft/world/map/MapDecoration
...
-	f	Lnet/minecraft/C_81psrrogw;	f_34tilfwue		f_34tilfwue
-	f	Lcom/google/common/base/Predicate;	f_59uavvhbh		f_59uavvhbh
-	f	Lnet/minecraft/C_64qcggszu$C_11itnlavb;	f_92pfgmcev		f_92pfgmcev
-	f	Lnet/minecraft/C_97zulxhng;	f_23jkexjie		f_23jkexjie
-	f	Ljava/lang/Class;	f_83kysuytu		f_83kysuytu
+c	net/minecraft/C_16xvdchlc		net/minecraft/entity/ai/goal/MobEntityActiveTargetGoal
+	c		Used exclusively by SlimeEntity in 1.8, essentially a variation of ActiveTargetGoal for non path finding entities
+	f	Lorg/apache/logging/log4j/Logger;	f_96zqstynf		LOGGER
+	f	Lnet/minecraft/C_81psrrogw;	f_34tilfwue		mob
+	f	Lcom/google/common/base/Predicate;	f_59uavvhbh		targetEntityFilter
```
Comment (SpaceWalkerRS) [inline mappings/14w27b#14w28a.tinydiff:2630]: `targetFilter`

File: mappings/14w27b#14w28a.tinydiff [hunk 12]
```diff
@@ -2623,21 +2623,22 @@ c	net/minecraft/C_16bwpyyiv		net/minecraft/world/map/MapDecoration
...
+	f	Lnet/minecraft/C_64qcggszu$C_11itnlavb;	f_92pfgmcev		entityDistanceComparator
+	f	Lnet/minecraft/C_97zulxhng;	f_23jkexjie		targetEntity
```
Comment (SpaceWalkerRS) [inline mappings/14w27b#14w28a.tinydiff:2632]: `target`

File: mappings/14w27b#14w28a.tinydiff [hunk 13]
```diff
@@ -2648,7 +2649,7 @@ c	net/minecraft/C_16xvdchlc$1		1
 	m	(Lnet/minecraft/C_97zulxhng;)Z	m_69nlwvdxx		apply
 		p	1			p_1
 	m	(Ljava/lang/Object;)Z	apply		apply
-		p	1			p_1
+		p	1			targetEntity
```
Comment (SpaceWalkerRS) [inline mappings/14w27b#14w28a.tinydiff:2652]: `target`

File: mappings/15w38a#15w38b.tinydiff [hunk 14]
```diff
@@ -91,9 +91,9 @@ c	net/minecraft/C_46nyuprdc
 c	net/minecraft/C_47ldwddrb
 	m	(DZLnet/minecraft/C_68zcrzyxg;Lnet/minecraft/C_76varpwca;)V	m_64xomyper	onFall	
 		p	1		dy	
-		p	3		p_3	
-		p	4		p_4	
-		p	5		p_5	
+		p	3		landed	
+		p	4		block	
+		p	5		blockPos
```
Comment (SpaceWalkerRS) [inline mappings/15w38a#15w38b.tinydiff:96]: `pos`

**Review by SpaceWalkerRS — CHANGES_REQUESTED** [review]

**Review by SpaceWalkerRS — APPROVED** [review]

**Comment by SpaceWalkerRS** [comment]
conflicts have been resolved, once the build action completes I'll merge

**Pull request merged**